### PR TITLE
FIX: fixed all the broken pages.

### PR DESCRIPTION
**Pull request by zoef-yt**
Fixed all the broken features, and pages and added suggestions from the previous branch.

### CODE REVIEW

**Review by srishtigautam21 — APPROVED** [review]
good to merge

File: src/Components/Modal/ModalComponent/CreatePostsCard.jsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { useDispatch, useSelector } from 'react-redux';
 import { closeModal } from '../../../store/Modal/ModalSlice';
 export const CreatePostsCard = () => {
 	const { authUser } = useSelector((store) => store.auth);
+	console.log(authUser);
```
Comment (dev-prabhat) [inline src/Components/Modal/ModalComponent/CreatePostsCard.jsx]: remove console.log

File: src/Firebase/FirebaseAuth.js [hunk 2]
```diff
@@ -132,6 +133,7 @@ const googleAuthHandler = async (errorHandler, setIsLoading, dispatch) => {
 		await signInWithPopup(auth, provider);
 		const currentUser = auth.currentUser;
 		const isUserInDb = await getSpecificUser(auth.currentUser.uid);
+		console.log(isUserInDb);
```
Comment (dev-prabhat) [inline src/Firebase/FirebaseAuth.js]: remove console.log

File: src/store/AllUser/AllUserSlice.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ const allUserSlice = createSlice({
 	initialState,
 	reducers: {
 		setAllUser: (state, action) => {
+			console.log(action.payload);
```
Comment (dev-prabhat) [inline src/store/AllUser/AllUserSlice.js]: remove console.log

**Review by dev-prabhat — APPROVED** [review]
Really like the UI part of this app, Just remove all console.log() from this PR
Else code structure is good

File: src/Components/Navbar/Navbar.css [hunk 4]
```diff
@@ -50,6 +50,52 @@
...
+
+.searchbar-holder .text-field {
+	margin: 1rem 0;
+}
+
+.searchbar-holder input:focus + .search-data {
+	height: min-content;
+	max-height: 30rem;
+	transform: scaleX(1);
+	transform-style: preserve-3d;
```
Comment (muskanbagrecha) [inline src/Components/Navbar/Navbar.css:65]: Nice, learnt something new.

File: src/Components/Navbar/Navbar.jsx [hunk 5]
```diff
@@ -32,3 +33,31 @@ const Navbar = () => {
 };
 
 export { Navbar };
+
+const SearchBar = () => {
+	const navigate = useNavigate();
+	const { allUsers } = useSelector((store) => store.allUsers);
+	const [search, setSearch] = useState('');
+	const searchedData = search !== '' ? allUsers.filter((user) => user.displayName.toLowerCase().includes(search.toLowerCase())) : allUsers;
```
Comment (muskanbagrecha) [inline src/Components/Navbar/Navbar.jsx:41]: In future, you can use debouncing for search.

File: src/Components/SinglePostCard/AllPostComments.jsx [hunk 6]
```diff
@@ -1,16 +1,25 @@
...
+	const { authUser } = useSelector((state) => state.auth);
+	const navigate = useNavigate();
 	const addComment = async () => {
-		if (userComment) {
-			await addCommentsToPost(postId, userId, userComment);
-			setUserComment('');
-			setTriggerEffect(!triggerEffect);
+		if (authUser) {
+			if (userComment) {
+				console.log(userComment);
```
Comment (muskanbagrecha) [inline src/Components/SinglePostCard/AllPostComments.jsx]: Remove console.log

File: src/Components/SinglePostCard/AllPostComments.jsx [hunk 7]
```diff
@@ -1,16 +1,25 @@
...
+				await addCommentsToPost(postId, userId, userComment);
+				setUserComment('');
+				setTriggerEffect(!triggerEffect);
+			}
+		} else {
+			navigate('/auth');
 		}
 	};
 	useEffect(() => {
```
Comment (muskanbagrecha) [inline src/Components/SinglePostCard/AllPostComments.jsx:24]: Please add dependency in useeffect

File: src/Components/SinglePostCard/AllPostComments.jsx [hunk 8]
```diff
@@ -20,6 +29,7 @@ const AllPostComments = ({ closeComments, commentSection, photoURL, displayName,
 
 	const getComments = async () => {
 		const comments = await allCommentsFromAPost(postId);
+		console.log(comments);
```
Comment (muskanbagrecha) [inline src/Components/SinglePostCard/AllPostComments.jsx]: remove console.log

File: src/Components/SinglePostCard/AllPostComments.jsx [hunk 9]
```diff
@@ -1,16 +1,25 @@
 import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
 import { addCommentsToPost, allCommentsFromAPost } from '../../Firebase/FirebaseFirestore';
 import { SingleComment } from './SingleComment';
 
 const AllPostComments = ({ closeComments, commentSection, photoURL, displayName, postId, userId }) => {
 	const [userComment, setUserComment] = useState('');
 	const [allComments, setAllComments] = useState([]);
 	const [triggerEffect, setTriggerEffect] = useState(false);
```
Comment (muskanbagrecha) [inline src/Components/SinglePostCard/AllPostComments.jsx:10]: Need to confirm if it is ideal to use a state only to trigger an effect.

File: src/Components/SinglePostCard/AllPostComments.jsx [hunk 10]
```diff
@@ -46,9 +56,7 @@ const AllPostComments = ({ closeComments, commentSection, photoURL, displayName,
 			</div>
 
 			{allComments.length > 0 ? (
-				allComments.map((comment) => (
-					<SingleComment comment={comment.comment} photoURL={photoURL} displayName={displayName} timestamp={comment.timestamp} />
-				))
+				allComments.map((comment) => <SingleComment comment={comment.comment} uid={comment.from} timestamp={comment.timestamp} />)
```
Comment (muskanbagrecha) [inline src/Components/SinglePostCard/AllPostComments.jsx]: Could destructure here.

File: src/Components/SinglePostCard/SinglePostCard.jsx [hunk 11]
```diff
@@ -13,6 +22,7 @@ const SinglePostCard = ({ caption, image, uid, index = 0, postId, timestamp }) =
 	const [edit, setEdit] = useState({ isEditing: false, caption: caption });
 	const openComments = () => setCommentSection(true);
 	const closeComments = () => setCommentSection(false);
+	const navigate = useNavigate();
 	const getDelay = (index) => {
```
Comment (muskanbagrecha) [inline src/Components/SinglePostCard/SinglePostCard.jsx:26]: Love this logic for animation.

File: src/Firebase/FirebaseAuth.js [hunk 12]
```diff
@@ -77,6 +77,7 @@ const emailSignUpHandler = async (errorHandler, setIsLoading, dispatch, setTextF
 			photoURL: defaultImage,
 		});
 		const currentUser = auth.currentUser;
+		console.log(currentUser);
```
Comment (muskanbagrecha) [inline src/Firebase/FirebaseAuth.js]: Remove console.log

File: src/Pages/ExplorePage/ExplorePage.jsx [hunk 13]
```diff
@@ -1,25 +1,61 @@
...
 	const { allPosts } = useSelector((store) => store.allPosts);
-
+	const filterBy = ['New', 'Old', 'Trending'];
+	const [selectedFilter, setSelectedFilter] = useState(filterBy[0]);
+	const optionsHandler = (e) => {
+		setSelectedFilter(e.target.value);
+	};
+	const getSortedData = (data) => {
+		if (allPosts) {
+			console.log('Inside this');
```
Comment (muskanbagrecha) [inline src/Pages/ExplorePage/ExplorePage.jsx]: Remove console.log

File: src/Pages/Home/HomePage.jsx [hunk 14]
```diff
@@ -10,8 +10,9 @@ const Home = () => {
 	const { allUsers } = useSelector((store) => store.allUsers);
 	const { allPosts } = useSelector((store) => store.allPosts);
 	const [homePageContent, setHomePageContent] = useState([]);
-	const currentUserHome = allPosts.filter((post) => following.findIndex((user) => user.id === post.uid) !== -1);
+	const currentUserHome = allPosts.filter((post) => following.findIndex((user) => user.id === post.uid || authUser?.uid === post.uid) !== -1);
 	useEffect(() => {
+		console.log('allPosts', currentUserHome);
```
Comment (muskanbagrecha) [inline src/Pages/Home/HomePage.jsx]: remove console.log

File: src/Pages/Home/HomePage.jsx [hunk 15]
```diff
@@ -27,15 +28,18 @@ const Home = () => {
 			</div>
 			<div>
 				{allPosts && homePageContent.length > 0 ? (
-					homePageContent.map((post) => {
+					homePageContent.map((post, index) => {
+						console.log(post);
```
Comment (muskanbagrecha) [inline src/Pages/Home/HomePage.jsx]: Remove console.log

File: src/Pages/ProfilePage/Components/UserDetails.jsx [hunk 16]
```diff
@@ -1,16 +1,18 @@
...
 import { setAuthUser } from '../../../store/Auth/AuthSlice';
 import { openModal } from '../../../store/Modal/ModalSlice';
 
-const UserDetails = () => {
+const UserDetails = ({ uid }) => {
 	const dispatch = useDispatch();
-	const { authUser, followers, following } = useSelector((store) => store.auth);
-	const { bio, displayName, photoURL, url } = authUser ?? { bio: '', displayName: '', photoURL: '', url: '' };
+	const navigate = useNavigate();
+	console.log(uid);
```
Comment (muskanbagrecha) [inline src/Pages/ProfilePage/Components/UserDetails.jsx]: Remove console.log

File: src/Pages/ProfilePage/Components/UserDetails.jsx [hunk 17]
```diff
@@ -19,38 +21,105 @@ const UserDetails = () => {
...
+			} else if (uid === authUser?.uid) {
+				setCurrentUser((prev) => ({ ...prev, followers: authUserFollowers.length, followings: authUserFollowings.length }));
+			}
+		}
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [allPosts, authUser?.uid]);
+	}, [uid, authUserFollowings]);
 
+	const followHandler = async (currentUserId, toFollowId) => {
+		console.log(authUser);
```
Comment (muskanbagrecha) [inline src/Pages/ProfilePage/Components/UserDetails.jsx]: Remove console.log

**Review by muskanbagrecha — CHANGES_REQUESTED** [review]
Great, abstracted code. Just remove the console.logs.

File: src/Components/Navbar/Navbar.jsx (hunk 5 above)
Comment (zoef-yt) [reply src/Components/Navbar/Navbar.jsx:41]: Sure will look into it

File: src/Components/SinglePostCard/AllPostComments.jsx (hunk 6 above)
Comment (zoef-yt) [reply src/Components/SinglePostCard/AllPostComments.jsx]: done

File: src/Components/SinglePostCard/AllPostComments.jsx (hunk 9 above)
Comment (zoef-yt) [reply src/Components/SinglePostCard/AllPostComments.jsx:10]: Okay will confirm this.

File: src/Components/SinglePostCard/AllPostComments.jsx (hunk 8 above)
Comment (zoef-yt) [reply src/Components/SinglePostCard/AllPostComments.jsx]: done

File: src/Components/SinglePostCard/SinglePostCard.jsx (hunk 11 above)
Comment (zoef-yt) [reply src/Components/SinglePostCard/SinglePostCard.jsx:26]: Thanks

File: src/Components/SinglePostCard/AllPostComments.jsx (hunk 10 above)
Comment (zoef-yt) [reply src/Components/SinglePostCard/AllPostComments.jsx]: done

File: src/Firebase/FirebaseAuth.js (hunk 12 above)
Comment (zoef-yt) [reply src/Firebase/FirebaseAuth.js]: done

File: src/Pages/ExplorePage/ExplorePage.jsx (hunk 13 above)
Comment (zoef-yt) [reply src/Pages/ExplorePage/ExplorePage.jsx]: done

File: src/Pages/Home/HomePage.jsx (hunk 14 above)
Comment (zoef-yt) [reply src/Pages/Home/HomePage.jsx]: done

File: src/Pages/Home/HomePage.jsx (hunk 15 above)
Comment (zoef-yt) [reply src/Pages/Home/HomePage.jsx]: done

File: src/Pages/ProfilePage/Components/UserDetails.jsx (hunk 16 above)
Comment (zoef-yt) [reply src/Pages/ProfilePage/Components/UserDetails.jsx]: done

File: src/Pages/ProfilePage/Components/UserDetails.jsx (hunk 17 above)
Comment (zoef-yt) [reply src/Pages/ProfilePage/Components/UserDetails.jsx]: done

File: src/store/AllUser/AllUserSlice.js (hunk 3 above)
Comment (zoef-yt) [reply src/store/AllUser/AllUserSlice.js]: done

File: src/Firebase/FirebaseAuth.js (hunk 2 above)
Comment (zoef-yt) [reply src/Firebase/FirebaseAuth.js]: done

File: src/Components/Modal/ModalComponent/CreatePostsCard.jsx (hunk 1 above)
Comment (zoef-yt) [reply src/Components/Modal/ModalComponent/CreatePostsCard.jsx]: done

**Review by muskanbagrecha — APPROVED** [review]
Code looks good to merge as you have incorporated the suggested changes. Great work on the project. Right from the UI, animations to the functionality. In future, you can extract logic outside the components. Other than that, everything looks good. Great effort.

**Pull request merged**